File: fix_rules.py
import re

with open('firestore.rules', 'r') as f:
    content = f.read()

pattern = re.compile(r"    match /notifications/\{notificationId\} \{\n      allow read: if signedIn\(\);\n      allow create: if isSuperAdmin\(\) \|\| \(signedIn\(\) && userExists\(\) && isManager\(userData\(\)\.companyId\)\);\n      allow update, delete: if isSuperAdmin\(\) \|\| \(signedIn\(\) && userExists\(\) && isCompanyAdmin\(userData\(\)\.companyId\)\);\n    \}")
content = pattern.sub("", content)

with open('firestore.rules', 'w') as f:
    f.write(content)

File: fix-rules.ts
import * as fs from 'fs';
const file = 'firestore.rules';
let content = fs.readFileSync(file, 'utf8');

const rfqRules = `
      // ==========================================
      // MODULE 14.2: RFQ MANAGEMENT SYSTEM
      // ==========================================
      match /rfq_requests/{rfqId} {
        allow read: if isSuperAdmin() || sameCompany(companyId);
        allow create, update: if isSuperAdmin() || isCompanyAdmin(companyId) || isManager(companyId);
        allow delete: if isSuperAdmin() || isCompanyAdmin(companyId);
      }
      match /rfq_bids/{bidId} {
        allow read: if isSuperAdmin() || sameCompany(companyId);
        allow create, update: if isSuperAdmin() || isCompanyAdmin(companyId) || isManager(companyId);
        allow delete: if isSuperAdmin() || isCompanyAdmin(companyId);
      }
      match /rfq_evaluation_logs/{logId} {
        allow read: if isSuperAdmin() || sameCompany(companyId);
        allow write: if isSuperAdmin() || isCompanyAdmin(companyId) || isManager(companyId);
      }
`;

if (!content.includes('rfq_requests')) {
  content = content.replace(
    '// ==========================================\n      // MODULE 14.1',
    rfqRules + '\n      // ==========================================\n      // MODULE 14.1'
  );
  fs.writeFileSync(file, content);
  console.log('Added RFQ Rules');
} else {
  console.log('RFQ Rules already exist');
}
